src/Sulu/Component/Content/Types/Link.php

```diff
@@ -111,7 +111,7 @@ public function getContentData(PropertyInterface $property): ?string
             return null;
         }
 
-        $url = \reset($linkItems)->getUrl();
+        $url = \end($linkItems)->getUrl();
```

Comment:
The array here is based on on the call in Line `108` so we are giving one `href` in it and we should get one href out of it.

So it sounds like an issue with one of our LinkProviders if it returns for a single `href` multiple results. Which one is failing in which cases?

Reply:
I see its the Media provider it looks like here is missing something:

https://github.com/sulu/sulu/blob/9361c365670e4b540897354b1c104ced1b0d0368/src/Sulu/Bundle/MediaBundle/Entity/MediaRepository.php#L219

This should join the current version `file.version = fileVersion.version` and return all versions.

Reply:
All media != all file versions we are returning beside the version history in all cases only the last active version. So it fine to filter here also by the actual version.

Reply:
ok, fixed, please review.